NEWS.md

```diff
@@ -40,6 +40,10 @@
         is changed to `$versions$r_package` (#791).
     -   `$rust_polars`, which indicates the version of the dependent Rust Polars,
         is changed to `$versions$rust_crate` (#791).
+-   New behavior when creating a `DataFrame` with a single list-variable. 
+    `pl$DataFrame(x = list(1:2, 3:4))` used to create a `DataFrame` with two
+    columns named "new_column" and "new_column_1", which was unexpected. It now
+    produces a `DataFrame` with a single `list` variable (#794).
```

Comment:
I suppose it affects to `$with_columns` etc. too.
Could you check them?

Reply:
Thanks I added some tests

Reply:
Thanks, could you please mention in the NEWS the other functions that are affected?

Reply:
done